dbt/adapters/bigquery/impl.py

```diff
@@ -115,6 +115,7 @@ class BigqueryConfig(AdapterConfig):
     max_staleness: Optional[str] = None
     enable_list_inference: Optional[bool] = None
     intermediate_format: Optional[str] = None
+    allow_non_incremental_definition: Optional[bool] = None
```

Comment:
Should this be a default of False? Having `None` introduces a Falsey third state, and I'm not sure it's needed

Reply:
`None` values get removed from the config entirely whereas `False` or `True` will get set. Also, the BQ "default" is nothing, not `False`. See my comment [here](https://github.com/dbt-labs/dbt-bigquery/pull/1011#discussion_r1631575612).

Reply:
Great. That's something I'll remember for later with respect to Bigquery. We historically live in a True-False exclusive world but I agree it makes sense to do this here.

Reply:
Maybe a code comment at the top explaining this is a practice used by bigquery with bools in their api design would help, I agree that it's counterintuitive and will likely trip up folks in the future

Reply:
It looks like the practice in general is to default all values for `AdapterConfig` to `None` (likely to make optional entries in the config `dict` via [filter_null_values](https://github.com/dbt-labs/dbt-common/blob/e30f65f6b93eed0422295a4282a892cdfe4b59d5/dbt_common/utils/dict.py#L11-L12)). For example, [here](https://github.com/dbt-labs/dbt-snowflake/blob/efe6df3c2941f722ec3e7449e3c0e0c019c435bf/dbt/adapters/snowflake/impl.py#L26-L36) is the same thing for `dbt-snowflake`. We have `Optional[bool]`s there as well, all defaulted to `None`. We then have a second "defaulting" behavior when we hit `BigQueryOptionsConfig` below. I'll mark this as resolved, that as unresolved, and continue there.